packages/imperative/src/config/src/ProfileCredentials.ts

```diff
@@ -109,8 +116,7 @@ export class ProfileCredentials {
      * @returns False if not using teamConfig or there are no secure fields
      */
     private isTeamConfigSecure(): boolean {
-        if (this.mProfileInfo.getTeamConfig().api.secure.secureFields().length === 0) return false;
-        return true;
+        return this.mProfileInfo.getTeamConfig().api.secure.secureFields(!this.mOnlyCheckActiveLayer).length > 0;
```

Comment:
As per the requested changes, the default behavior of this function now looks at all layers to determine whether credentials are stored securely.

To revert this behavior, application developers (like Zowe Explorer for VSCode and its extenders) can simply initialize the ProfileCredentials class with `onlyCheckActiveLayer: true` 🙏